gobblin-compaction/src/main/java/gobblin/compaction/event/CompactionSlaEvent.java

```diff
@@ -0,0 +1,129 @@
+package gobblin.compaction.event;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.Counter;
+import org.apache.hadoop.mapreduce.Counters;
+import org.apache.hadoop.mapreduce.Job;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Maps;
+
+import gobblin.compaction.mapreduce.avro.AvroKeyDedupReducer;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.metrics.event.sla.SlaEvent;
+import gobblin.metrics.event.sla.SlaEventKeys;
+
+
+public class CompactionSlaEvent extends SlaEvent {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CompactionSlaEvent.class);
+  private static final String EVENT_NAME = "CompactionCompleted";
+  private static final String DEFAULT = StringUtils.EMPTY;
+
+  private Job job;
```

Comment:
You can still use the Job/state objects to instantiate the Event. What I am saying is that event should not keep those pointers.